### PR TITLE
REF: dont alter state in pytables read_axes

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/io/pytables.py [hunk 1]
```diff
@@ -2354,8 +2358,20 @@ def validate_attr(self, append):
...
+        Parameters
+        ----------
+        values : np.ndarray
+        nan_rep :
+        encoding : str
+        errors : str
+
+        Returns
+        -------
+        index : listlike to become an Index
```
Comment (jreback) [inline pandas/io/pytables.py]: ideally type at some point (e.g. return value)

File: pandas/io/pytables.py [hunk 2]
```diff
@@ -4334,34 +4349,51 @@ def read(
...
+
         info = (
             self.info.get(self.non_index_axes[0][0], dict())
             if len(self.non_index_axes)
             else dict()
         )
-        index = self.index_axes[0].values
+
+        axes = list(self.axes)
+        inds = [i for i in range(len(axes)) if axes[i] is self.index_axes[0]]
```
Comment (jreback) [inline pandas/io/pytables.py]: can you just inline axes

File: pandas/io/pytables.py [hunk 3]
```diff
@@ -4334,34 +4349,51 @@ def read(
...
+        assert len(inds) == 1
+        ind = inds[0]
+
+        index = result[ind][0]
+
         frames = []
-        for a in self.values_axes:
+        for i, a in enumerate(self.axes):
```
Comment (jreback) [inline pandas/io/pytables.py]: confusing y ou are using self.axes here again

**Review by jreback — CHANGES_REQUESTED** [review]
would rebase

File: pandas/io/pytables.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/io/pytables.py]: yah, ATM we're not consistent about what this is (e.g. sometimes list) so id like to to get this more specific

File: pandas/io/pytables.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/io/pytables.py]: sure

File: pandas/io/pytables.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/io/pytables.py]: we need `i` to do the lookup `result[i]` and `a` to do the check just below.  there may be a nicer way to do this if/when `values_axes` ceases to be stateful

**Comment by jbrockmendel** [comment]
rebased+green

**Comment by jbrockmendel** [comment]
Not sure whats going on here
```
/home/vsts/work/1/s/pandas/core/indexes/multi.py:docstring of pandas.MultiIndex.rst:91: WARNING: autosummary: stub file not found 'pandas.MultiIndex.get_locs'. Check your autosummary_generate setting.
```

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**